parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -919,6 +895,59 @@ public void appendRowGroup(SeekableInputStream from, BlockMetaData rowGroup,
     endBlock();
   }
 
+  /**
+   * Merges adjacent row groups in the supplied files while maintaining that the new groups is no more than the specified
+   * maxRowGroupSize
+   * @param inputFiles input files to merge
+   * @param maxRowGroupSize the maximum size in bytes the new created groups can be
+   * @param useV2Writer whether to use a V2 encoding based writer when rewriting dictionary encoded pages
+   * @param compression compression to use when writing
+   * @throws IOException
+   */
+  public void mergeRowGroups(List<InputFile> inputFiles, long maxRowGroupSize, boolean useV2Writer, CompressionCodecName compression) throws IOException {
```

Comment:
I am not sure we want to release a public API to PartquetFileWriter which is heavily used.

Reply:
That is a good question. My thinking is that this method is similar(an improvement) to _appendRowGroup_ which is a public API. In most use cases the user of _appendRowGroup_ actually wants the functionality provided by this method.  But it boils down to if _appendRowGroup_ really belonged here in the first place. I have no strong position on this, do you think I should not expose this new method?

Reply:
I prefer not to unless you strongly think we should.